### ISSUE TITLE
Advanced Javascript - Async Code: Add comparison between callback and Promise, video format 

### ISSUE THREAD

**Issue by i5x64BIT** [TheOdinProject/curriculum#26150]
### Describe your suggestion

Add [this video](https://www.youtube.com/watch?v=ranuTFXPgbw) that explains why we use promises instead of callbacks.
It is mostly based on the `YDNJS: Chapter 3`  reffered in the article.

Appened to `Additional Resources`

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/node-path-javascript-asynchronous-code

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

**Comment by wise-king-sullyman** [comment]
Thanks for submitting this issue!

This lesson is already at the maximum of additional resources we generally want, and we already have an assignment and two additional resource videos about promises. 

Do you think this resource is worth removing one of the existing additional resources for?

If so please state which one and why you believe it's worth replacing with this video.

**Comment by i5x64BIT** [comment]
If any resource should be removed, it would be one of the 3 "Understanding Promises" videos. I understood promises by just giving some time to the YDNJS, yet the topic of Callbacks approach compared to Promises is not covered enough, in my opinion.

**Comment by wise-king-sullyman** [comment]
Which of the 3 do you think is the weakest?

**Comment by i5x64BIT** [comment]
I would say `Understanding Promises` by Digital Ocean. All points are already covered, in written format by the book.

**Comment by wise-king-sullyman** [comment]
I'm a bit confused, I don't see a video by Digital Ocean?

**Comment by i5x64BIT** [comment]
Hi, I am sorry for the confusion. I attended the issue in a rush.
After a carefull read I would say that both the articles and begginer videos are strong resources indeed.
Yet, There are lessons in the curriculum that hold more then 4 informational resources at the bottom, moreover, from the discussion you can infer that a resource on callbacks vs promises is indeed missing (other then the book of cource).
Therefore I belive this video deserves an extra spot, since it is also a strong one.

**Comment by i5x64BIT** [comment]
As you would probably agree, understanding the "why" is as important as the "how"

**Comment by wise-king-sullyman** [comment]
Yes, some lessons have many more additional resources, but this is an area we're consciously trying to keep from growing extraneously going forward as we've received feedback that a significant amount of learners try to go through all additional resources.

This is why we've updated our[ layout guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#lesson-layout) to specify that lessons should have no more than 3-5 additional resources at an absolute max, with >3 usually being reserved for great related content rather than just content that re-covers already covered information in an alternative format.

Since as you've said this video is mostly based on a reading already in our assignments I do not believe we should make an exception to the above guidelines in this case and I'm opposed to us adding it unless we replace one of the current additional resource videos.

**Comment by i5x64BIT** [comment]
Well I haven't tought of that to be honest, Often times I took a few days to compleate a module.
It might be just me, but I feel that there is not enough video material to compliment the written stuff.
Not just here but all the path section, and I'm not sure if that's the whole idea or it's just unnecessary overreliance on books and blog posts.
For me spoken and video just click much more and I find myself way to often looking for a youtube video to reinforce what I've just read.

But all&all, fair point @wise-king-sullyman

**Comment by wise-king-sullyman** [comment]
Yeah we often favor written material over video. In part because it's more maintainable, in part because it's usually more accessible, and in part because we believe it's often more effective. If you're curious I can go into more detail about our reasonings for the above.

All that aside though I want to let you know that we do truly appreciate the time you took to make this 🙂
